### ISSUE TITLE
Keep current tree structure when reconnecting

### ISSUE THREAD

**Issue by isherwood** [icetee/remote-ftp#408]
I experience regular disconnects. I'm not sure whether it's my config or my server setup, but I often have to manually connect to get files to save. I'd like for the tree to not collapse when I do so, requiring me to drill back into my working directory (or multiple, in the case of my current project).

**Comment by brianjcohen** [comment]
+1 for this. Possibly related to #349 . The behavior I'm looking for is present in Komodo Edit, which is that connections are made implicitly as a result of the user attempting to interact with the remote objects. So if I click on a folder or double-click a file in the remote tree, but the connection has dropped, remote-ftp should first re-establish the connection and then proceed with showing me the contents of the folder or file.  Likewise if I attempt to save a file and the connection has dropped, remote-ftp should first re-establish the connection and then proceed with saving the file.  As it currently functions, the action (eg. open, save) is denied and the tree view is wiped out, forcing me to manually re-connect and manually re-try the action.

**Comment by samifouad** [comment]
It's hard to say whether this is a bug or not, but the desired persistence is definitely a good idea.

**Comment by qbadev** [comment]
I agree. "Drilling" really annoys. Manual reconnecting is also my issue as well as @brianjcohen. I also used to use Komodo and its FTP extension.
